src/context.js

```diff
@@ -16,12 +18,73 @@ class Context {
     this.dotenvs = this.dotenvs.concat(donenvs);
   }
 
-  clone(home = null) {
-    return module.exports.create((home)?home:this.home, this.name, this.uuid, this.argv, this.env, this.dotenvs, this.save, this.validate );
+  snapshotDotenvs() {
+    let dotenvs = [];
+    // find head of the chain
+    let head = this;
+    while (head.prev) {
+      head = head.prev;
+    }
+    // check if we have anchor chain
+    const anchor = head.anchor;
+    if (anchor) {
+      let tail = this;
+      while (tail.next) {
+        tail = tail.prev;
+      }
+        // insert current chain into anchor's chain
+      // const after = anchor;
+      const before = anchor.next;
+      //
+      anchor.next = head;
+      head.prev = anchor;
+      if (before) {
+        before.prev = tail;
+      }
+      tail.next = before;
+      // get dotenvs from anchor's chain
+      dotenvs = anchor.snapshotDotenvs();
+      // restore anchors chain
+      anchor.next = before;
+      if (before) {
+        before.prev = anchor;
+      }
+      head.prev = null;
+      tail.next = null;
+    } else {
+      // examine all items in chain
+      do {
+        dotenvs = dotenvs.concat(head.dotenvs);
+        head = head.next;
+      } while (head);
+      dotenvs = dotenvs.reverse();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove or correct this useless self-assignment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://159.65.230.204:9000/coding_rules#rule_key=javascript%3AS1656)